GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -2713,14 +2713,29 @@ private void UpdateAuthorInfo()
             ThreadHelper.FileAndForget(async () =>
                 {
                     // Do not cache results in order to update the info on FormActivate
-                    string userName = Module.GetEffectiveGitSetting(SettingKeyString.UserName, cache: false);
```

Comment:
Do not try-catch if the reason is known - in this case git-log return 0 is OK
This is the part of #11146 I approved (the rest of the discussion there just got me loose motivation for GE and a dev in general though).

Could be an option to GetEffectiveGitSetting, but this is currently the only caller.

It is also not necessary to call UpdateAuthorInfo() in OnActivate.

[0001-Do-not-throw-for-git-log-no-key-exists.patch](https://github.com/gitextensions/gitextensions/files/13401403/0001-Do-not-throw-for-git-log-no-key-exists.patch)

Reply:
Thank you for the patch. I did not think of its availability when I decided to not touch the `ExecutableExtensions.GetOutput()` chain.

> It is also not necessary to call UpdateAuthorInfo() in OnActivate.

The setting could be changed using another app (instance).
But removing `OnActivate` was the easiest way to get rid of repeated exception popups.

Reply:
Look at  the status bar. On activate is on purpose so that the commiter and author info is updated when user activates window because git config could have been executed. It is to make sure user sees up to date info.

Reply:
>  It is to make sure user sees up to date info.

As said it is not worth it. If users use another app / instance for changing the settings, they can reopen FormCommit if they wonder whether the settings are in effect or not. That's not that unusual.

Reply:
I agree with @mstv, it's not worth it.